src/mjolnir/graphenhancer.cc

```diff
@@ -539,6 +826,82 @@ bool IsIntersectionInternal(const GraphTile* start_tile,
   return true;
 }
 
+// Get the Headings for the node.
+void GetHeadings(GraphTileBuilder& tile, NodeInfo& nodeinfo, uint32_t ntrans) {
+  if (ntrans == 0) {
+    throw std::runtime_error("edge transitions set is empty");
+  }
+
+  std::vector<uint32_t> heading(ntrans);
+  nodeinfo.set_local_edge_count(ntrans);
+  for (uint32_t j = 0; j < ntrans; j++) {
+    DirectedEdge& de = tile.directededge_builder(nodeinfo.edge_index() + j);
+
+    auto e_offset = tile.edgeinfo(de.edgeinfo_offset());
+    auto shape = e_offset.shape();
+    if (!de.forward()) {
+      std::reverse(shape.begin(), shape.end());
+    }
+    heading[j] = std::round(
+        PointLL::HeadingAlongPolyline(shape, GetOffsetForHeading(de.classification(), de.use())));
+
+    // Set heading in NodeInfo. TODO - what if 2 edges have nearly the
+    // same heading - should one be "adjusted" so the relative direction
+    // is maintained.
+    nodeinfo.set_heading(j, heading[j]);
+  }
+}
+
+// Is the next edge from the end node of the directededge is internal or not.
+bool IsNextEdgeInternal(const DirectedEdge directededge,
```

Comment:
Used to determine if we delete the lane info on the prior edge or on the last edge